### PR TITLE
Convert our CI files to resuable workflows

### PR DESCRIPTION
**Pull request by genebean**
See https://docs.github.com/en/actions/using-workflows/reusing-workflows

### CODE REVIEW

**Comment by genebean** [comment]
Fixing docs, one sec

**Comment by yachub** [comment]
Hooray! Works in https://github.com/ploperations/ploperations-unicorn/pull/19

File: .github/workflows/pr_test.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        puppet_versions:
+          - '7'
+          - '8'
+
+    name: Puppet ${{ matrix.puppet_versions }}
+    steps:
+    - uses: actions/checkout@v1
```
Comment (yachub) [inline .github/workflows/pr_test.yml]: I think that v1 and v2 are deprecated
```suggestion
    - uses: actions/checkout@v3
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+# In the repo that uses this workflow, this should be set to
+# on: workflow_dispatch
+
+jobs:
+  publish-module:
+    name: Tag Release and Publish to Forge
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
```
Comment (yachub) [inline .github/workflows/release.yml]: I think that v1 and v2 are deprecated
```suggestion
        uses: actions/checkout@v3
```

**Review by yachub — APPROVED** [review]

**Pull request merged**